scheds/rust/scx_rusty/src/bpf/main.bpf.c

```diff
@@ -440,32 +505,225 @@ static void refresh_tune_params(void)
+	 * deadline accordingly. Much of this is inspired by the logic in
+	 * scx_lavd that was originally conceived and implemented by Changwoo
+	 * Min. Though the implementations for determining latency criticality
+	 * are quite different in many ways, individuals familiar with both
+	 * schedulers will feel an eerie sense of deja-vu. The details of
+	 * interactivity boosting for rusty are described below.
+	 */
+
+	/*
+	 * We begin by calculating the following interactivity factors for a
+	 * task:
+	 *
+	 * - waker_freq: The frequency with which a task wakes up other tasks.
+	 *		 A high waker frequency generally implies a producer
+	 *		 task that is at the beginning and/or middle of a work
+	 *		 chain.
+	 *
+	 * - blocked_freq: The frequency with which a task is blocked. A high
+	 *		   blocked frequency indicates a consumer task that is
+	 *		   at the middle and/or end of a work chain.
+	 *
+	 * A task that is high in both frequencies indicates what is often the
+	 * most latency-critical interactive task: a task that functions both
+	 * as a producer and a consumer by being in the _middle_ of a work
+	 * chain.
+	 *
+	 * We want to prioritize running these tasks, as they are likely to
+	 * have a disproporionate impact on the latency (and possibly
+	 * throughput) of the workload they are enabling due to Amdahl's law.
+	 * For example, say that you have a workload where 50% of the workload
+	 * is serialized by a producer and consumer task (25% each), and the
+	 * latter 50% is serviced in parallel by n CPU hogging tasks. If either
+	 * the producer or consumer is improved by a factor of x, it improves
+	 * the latency of the entire workload by:
+	 *
+	 *	S_lat(x) = 1 / ((1 - .25) + (.25 / x))
+	 *
+	 * Say that we improve wakeup latency by 2x for either task, the
+	 * latency improvement would be:
+	 *
+	 *	S_lat(2) = 1 / ((1 - .25) + (.25 / 2))
+	 *		 = 1 / (.75 + .125)
+	 *		 = 1 / .875
+	 *		~= 14.2%
+	 *
+	 * If we instead improve wakeup latency by 2x for all the n parallel
+	 * tasks in the latter 50% of the workload window, the improvement
+	 * would be:
+	 *
+	 *	S_lat(2) = 1 / ((1 - .5) + (.5 / 2))
+	 *		 = 1 / (.5 + .25)
+	 *		 = 1 / .75
+	 *		~= 33%
+	 *
+	 * This is also significant, but the returns are amortized across all
+	 * of those tasks. Thus, by giving a latency boost to the producer /
+	 * consumer tasks, we optimize for the case of scheduling tasks that
+	 * are on the critical path for serial workchains, and have a
+	 * disproporionate impact on the latency of a workload.
+	 *
+	 * We multiply the frequencies of wait_freq and waker_freq somewhat
+	 * arbitrarily, based on observed performance for audio and gaming
+	 * interactive workloads.
+	 */
+	waker_freq = max_clamp(taskc->waker_freq, DL_FREQ_FT_MAX);
+	blocked_freq = max_clamp(taskc->blocked_freq, DL_FREQ_FT_MAX);
+	freq_factor = blocked_freq * waker_freq * waker_freq;
+	freq_factor = scale_fair(freq_factor, p->scx.weight);
```

Comment:
Definitely, I'll add it. Your comment also made me realize that I meant to scale `freq_factor` _positively_ by weight rather than negatively (which we do want to do for `avg_runtime`. So I'll not only add a comment, but also test changing this to scale positively with weight to see what the effect is. Thanks!